programs/locker/src/instructions/v2/create_vesting_escrow.rs

```diff
@@ -0,0 +1,114 @@
+use anchor_spl::token::spl_token;
+use anchor_spl::token_2022::spl_token_2022;
+use anchor_spl::token_interface::{Mint, TokenAccount, TokenInterface};
+
+use crate::util::{calculate_transfer_fee_included_amount, transfer_to_escrow_v2, validate_mint};
+use crate::TokenProgramFlag::{UseSplToken, UseToken2022};
+use crate::*;
+
+#[event_cpi]
+#[derive(Accounts)]
+pub struct CreateVestingEscrowV2<'info> {
+    #[account(mut)]
+    pub base: Signer<'info>,
+
+    #[account(
+        init,
+        seeds = [
+        b"escrow".as_ref(),
+        base.key().as_ref(),
+        ],
+        bump,
+        payer = sender,
+        space = 8 + VestingEscrow::INIT_SPACE
+    )]
+    pub escrow: AccountLoader<'info, VestingEscrow>,
+
+    pub token_mint: Box<InterfaceAccount<'info, Mint>>,
+
+    #[account(
+        mut,
+        associated_token::mint = sender_token.mint,
+        associated_token::authority = escrow,
+        associated_token::token_program = token_program
+    )]
+    pub escrow_token: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    #[account(mut)]
+    pub sender: Signer<'info>,
+
+    #[account(mut)]
+    pub sender_token: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    /// CHECK: recipient account
+    pub recipient: UncheckedAccount<'info>,
+
+    /// Token program.
+    pub token_program: Interface<'info, TokenInterface>,
+
+    // system program
+    pub system_program: Program<'info, System>,
+}
+
+pub fn handle_create_vesting_escrow_v2<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, CreateVestingEscrowV2<'info>>,
+    params: &CreateVestingEscrowParameters,
+) -> Result<()> {
+    require!(
+        validate_mint(&ctx.accounts.token_mint).unwrap(),
```

Comment:
Could function validate_mint() return Result<()>, then you can use validate_mint()?

Reply:
updated